system/lib/pthread/library_pthread.c

```diff
@@ -825,17 +825,26 @@ EMSCRIPTEN_KEEPALIVE void* _emscripten_main_thread_futex;
 
 void __emscripten_init_main_thread_js(void* tb);
 
+static void *dummy_tsd[1] = { 0 };
+weak_alias(dummy_tsd, __pthread_tsd_main);
+
 // See system/lib/README.md for static constructor ordering.
 __attribute__((constructor(48)))
 void __emscripten_init_main_thread(void) {
   __emscripten_init_main_thread_js(&__main_pthread);
+  //printf("__emscripten_init_main_thread %p\n", __main_pthread);
+
   // The pthread struct has a field that points to itself - this is used as
   // a magic ID to detect whether the pthread_t structure is 'alive'.
   __main_pthread.self = &__main_pthread;
+  __main_pthread.detach_state = DT_JOINABLE;
   // pthread struct robust_list head should point to itself.
   __main_pthread.robust_list.head = &__main_pthread.robust_list.head;
   // Main thread ID is always 1.  It can't be 0 because musl assumes
   // tid is always non-zero.
   __main_pthread.tid = getpid();
   __main_pthread.locale = &libc.global_locale;
+  __main_pthread.next = __main_pthread.prev = &__main_pthread;
+
+  __main_pthread.tsd = (void **)__pthread_tsd_main;
```

Comment:
Since `->tsd` is now always initialized, I think we can get rid of this check:
https://github.com/emscripten-core/emscripten/blob/a47616119653aa89542dfac531722520fb16f0af/system/lib/libc/musl/src/thread/pthread_getspecific.c#L7-L10
(See previous discussion here: https://github.com/emscripten-core/emscripten/pull/14775/files#r678491019)

Reply:
Yes, I think you are correct.